lightning/src/ln/channel.rs

```diff
@@ -1895,6 +1895,16 @@ impl<Signer: Sign> Channel<Signer> {
 			return Err(ChannelError::Close("Minimum confirmation depth must be at least 1".to_owned()));
 		}
 
+		if let Some(ty) = &msg.channel_type {
+			if *ty != self.channel_type {
+				return Err(ChannelError::Close("Channel Type in accept_channel didn't match the one sent in open_channel.".to_owned()));
```

Comment:
I think the spec says in `accept_channel` subsection:

"The sender if it sets the `channel_type` MUST set it to the `channel_type` from `open_channel`".

I agree the "it" isn't clear though sounds the spec requirement to echo is already there.

Reply:
I'd think `it` is pretty clearly (a) the sender, and (b) the `channel_type`, so technically I don't think we're in violation today, though we maybe should be.

Reply:
No, the spec very clearly does *not* explicitly require this - it says "IF [the sender] sets the `channel_type`"...then it must set it to a copy. The spec does not require the `accept_channel` sender to send a copy.